oap-server/server-receiver-plugin/skywalking-mq-receiver-plugins/kafka-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/kafka/provider/KafkaReceiverProvider.java

```diff
@@ -0,0 +1,116 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.configuration.api.ConfigurationModule;
+import org.apache.skywalking.oap.server.core.CoreModule;
+import org.apache.skywalking.oap.server.core.cluster.ClusterModule;
+import org.apache.skywalking.oap.server.library.module.ModuleConfig;
+import org.apache.skywalking.oap.server.library.module.ModuleDefine;
+import org.apache.skywalking.oap.server.library.module.ModuleProvider;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.apache.skywalking.oap.server.library.module.ServiceNotProvidedException;
+import org.apache.skywalking.oap.server.receiver.kafka.KafkaReceiveHandlerRegister;
+import org.apache.skywalking.oap.server.receiver.kafka.module.KafkaReceiverConfig;
+import org.apache.skywalking.oap.server.receiver.kafka.provider.handler.JVMMetricsReceiveHandler;
+import org.apache.skywalking.oap.server.receiver.kafka.provider.handler.KafkaManagementReceiveHandler;
+import org.apache.skywalking.oap.server.receiver.kafka.provider.handler.TraceSegmentReceiveHandler;
+import org.apache.skywalking.oap.server.receiver.mq.ReceiveHandlerRegister;
+import org.apache.skywalking.oap.server.receiver.mq.module.MessageQueueReceiverModule;
+import org.apache.skywalking.oap.server.receiver.sharing.server.SharingServerModule;
+import org.apache.skywalking.oap.server.receiver.trace.provider.TraceServiceModuleConfig;
+import org.apache.skywalking.oap.server.receiver.trace.provider.parser.SegmentParserListenerManager;
+import org.apache.skywalking.oap.server.receiver.trace.provider.parser.listener.MultiScopesAnalysisListener;
+import org.apache.skywalking.oap.server.receiver.trace.provider.parser.listener.NetworkAddressAliasMappingListener;
+import org.apache.skywalking.oap.server.receiver.trace.provider.parser.listener.SegmentAnalysisListener;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+
+@Slf4j
+public class KafkaReceiverProvider extends ModuleProvider {
+    private KafkaReceiverConfig config;
+    private KafkaReceiveHandlerRegister handlerRegister;
+    private TraceServiceModuleConfig traceServiceModuleConfig;
+
+    public KafkaReceiverProvider() {
+        config = new KafkaReceiverConfig();
+    }
+
+    @Override
+    public String name() {
+        return "kafka";
+    }
+
+    @Override
+    public Class<? extends ModuleDefine> module() {
+        return MessageQueueReceiverModule.class;
+    }
+
+    @Override
+    public ModuleConfig createConfigBeanIfAbsent() {
+        return config;
+    }
+
+    @Override
+    public void prepare() throws ServiceNotProvidedException, ModuleStartException {
+        handlerRegister = new KafkaReceiveHandlerRegister(config);
+        this.registerServiceImplementation(ReceiveHandlerRegister.class, handlerRegister);
+    }
+
+    @Override
+    public void start() throws ServiceNotProvidedException, ModuleStartException {
```

Comment:
I think once you are opening receiver, this should take the responsibility of creating topic(s), right?

Reply:
Kafka consumer/product will create it if not exists in default. I will recheck.